ruma-identifiers/src/opaque_ids.rs

```diff
@@ -28,6 +28,14 @@ opaque_identifier! {
     pub type KeyName;
 }
 
+opaque_identifier! {
+    /// The `media-id` of a Matrix Content (MXC) URI
+    ///
+    /// Media identifiers in Matrix are opaque character sequences of `[a-zA-Z_-]`. This type is provided
+    /// simply for its semantic value.
+    pub type MediaId;
```

Comment:
I'm not sure we need this extra type. But if we want to keep it, we should probably also enforce the character range.

Reply:
I added it since media requests expect the media id separate from the server name.

Reply:
i removed it.